app/wasm.go

```diff
@@ -110,3 +111,26 @@ func isStargateMsg(msg wasmvmtypes.CosmosMsg) bool {
 func isIBCSendPacketMsg(msg wasmvmtypes.CosmosMsg) bool {
 	return msg.IBC != nil && msg.IBC.SendPacket != nil
 }
+
+type WasmAppModuleBasicOverride struct {
+	wasm.AppModuleBasic
+	uploader sdk.AccAddress
+}
+
+func NewWasmAppModuleBasicOverride(wasmModule wasm.AppModuleBasic, uploader sdk.AccAddress) WasmAppModuleBasicOverride {
+	return WasmAppModuleBasicOverride{
+		AppModuleBasic: wasmModule,
+		uploader:       uploader,
+	}
+}
+
+// DefaultGenesis returns an override for the wasm module's DefaultGenesis,
+// because as soon as the module is initialized the restriction to contract upload and instantiation must hold
+func (m WasmAppModuleBasicOverride) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
+	return cdc.MustMarshalJSON(&wasm.GenesisState{
+		Params: wasmtypes.Params{
+			CodeUploadAccess:             wasmtypes.AccessTypeAnyOfAddresses.With(m.uploader),
```

Comment:
```suggestion
			CodeUploadAccess:             wasmtypes.AccessTypeOnlyAddress.With(m.uploader),
```
Why not use `AccessTypeOnlyAddress` instead?

Reply:
it's deprecated